Hamburger Menu

MatthiasReumann opened the pull request:
Closes #237 (Add hamburger menu on mobile devices)

MatthiasReumann committed:
fix minor ui bugs

MatthiasReumann committed:
Add hamburger

MatthiasReumann committed:
Update hamburger

alexanderstephan reviewed (changes requested):
Looks good to me codewise, design wise I also really like this!

When a played around with it, I noticed a couple minor things:
I noticed that somehow there is a lot free space in the menu in my browser, see the image blow. Resolution is 637x874 in this example.

<img src="https://user-images.githubusercontent.com/30267166/150431342-9f52b2a6-aeba-49b3-92ec-a2caddf6410c.png" width="300">
Maybe we can reduce the right margin? We could also think about putting the dark theme and GitHub icon directly below the other menu items, I guess.

I also think that in light mode, the menu items could be almost black instead of grey. I have a hard time reading the text here.

I noticed that the "Login" spacing is different on my Honor 6x Android 7 Phone on Firefox. I can't reproduce it on my Desktop though. So not sure, if this easily fixable or even a problem on my end.
<img src="https://user-images.githubusercontent.com/30267166/150432851-9935f851-a66d-43a2-9bda-d550dbe8678a.png" width="300">

joschahenningsen commented on web/template/header.gohtml:
```diff
@@ -1,56 +1,123 @@
+                        <a href="/login"
+                           class="font-semibold text-gray-400">
+                            <i class="fas fa-sign-in-alt w-8"></i>
+                            <span>Login</span>
```
```suggestion
                            <span class="ml-3">Login</span>
```

joschahenningsen replied:
This should fix @alexanderstephan s observation

joschahenningsen commented:
Other than what @alexanderstephan pointed out, this looks great! I agree that there is quite a bit of space here, but I don't mind.

MatthiasReumann committed:
Update mobile nav

MatthiasReumann committed:
Add 'no upcoming stream' to FriendlyNextDate function

MatthiasReumann commented:
> Looks good to me codewise, design wise I also really like this!
> 
> When a played around with it, I noticed a couple minor things: I noticed that somehow there is a lot free space in the menu in my browser, see the image blow. Resolution is 637x874 in this example.
> 
> <img alt="" width="300" src="https://user-images.githubusercontent.com/30267166/150431342-9f52b2a6-aeba-49b3-92ec-a2caddf6410c.png">
> 
> Maybe we can reduce the right margin? We could also think about putting the dark theme and GitHub icon directly below the other menu items, I guess.
> I also think that in light mode, the menu items could be almost black instead of grey. I have a hard time reading the text here.
> 
> I noticed that the "Login" spacing is different on my Honor 6x Android 7 Phone on Firefox. I can't reproduce it on my Desktop though. So not sure, if this easily fixable or even a problem on my end. <img alt="" width="300" src="https://user-images.githubusercontent.com/30267166/150432851-9935f851-a66d-43a2-9bda-d550dbe8678a.png">

- Fixed the Login bug (Thanks @joschahenningsen!) 
- Changed the dark-mode button to a toggle thingy 
- Moved the GitHub and toggle button up
- Light mode text black

MatthiasReumann committed:
Merge branch 'dev' into enhancement/hamburger

# Conflicts:
#	web/template/partial/stream/chat.gohtml

joschahenningsen reviewed (approved):
Looks super clean :fire:

alexanderstephan reviewed (approved):
Looks awesome, I also really like the addition of the dark mode toggle! :fire: Maybe this means we should close https://github.com/joschahenningsen/TUM-Live/pull/134?
